Scripts/script-IndicatorMaliciousRatioCalculation.yml

```diff
@@ -0,0 +1,110 @@
+      result['CustomFields'] = hash_object(result.get('CustomFields'))
+      return result
+
+
+  res = demisto.executeCommand("findIndicators", {'query': QUERY, 'size': MAX_INDICATORS})
+  indicators = res[0]['Contents']
+  indicators_result = map(get_indiator_data, indicators)
+  res = demisto.executeCommand("getIncidents", {'query': QUERY.replace('incident.', ''), 'size': MAX_INCIDENTS})
+  incidents = res[0]['Contents']['data']
+  if incidents:
+      incidents_result = map(get_incident_data, incidents)
+      resolved_incident_ids = set(map(lambda x: x['id'], incidents_result))
+      non_resolved_id = set()
+      for i in indicators_result:
+          resolved = [x for x in i['investigationIDs'] if x in resolved_incident_ids]
+          non_resolved_id = non_resolved_id.union(set(i['investigationIDs']).difference(set(resolved)))
+          resolved_count = len(resolved)
+          total_count = len(i['investigationIDs'])
+          i['resolved_incidents_count'] = resolved_count
+          i['total_incidents_count'] = total_count
+          i['malicious_ratio'] = float(resolved_count) / total_count
+
+      res = demisto.executeCommand("getIncidents", {'query': " or ".join(map(lambda x: "id:%s" % x, non_resolved_id)), 'size': MAX_INCIDENTS})
+      non_resolved_incidents = res[0]['Contents']['data']
+      non_resolved_incidents = map(get_incident_data, non_resolved_incidents)
+
+      indicators_result = [x for x in indicators_result if x['total_incidents_count'] >= APPEARS_IN_MIN_NUMBER_OF_INCIDENTS]
+      indicators_result.sort(key=lambda x: x['malicious_ratio'], reverse=True)
+      demisto.results({
+          'Type': entryTypes['note'],
+          'Contents': indicators_result,
+          'ContentsFormat': formats['json'],
+          'ReadableContentsFormat': formats['markdown'],
+          'HumanReadable': tableToMarkdown('Indicators Malicious Ratio', indicators_result[:50], headers=['value', 'indicator_type', 'malicious_ratio', 'total_incidents_count', 'score', 'lastSeen'])
+      })
+
+      for i in indicators_result:
+          i['value'] = hash_object(i['value'])
+
+      demisto.results(fileResult('MaliciousRatio.json', json.dumps({
+          'resolved_incidents': incidents_result,
+          'non_resolved_incidents': non_resolved_incidents,
+          'indicators': indicators_result
+      })))
+  else:
+      demisto.results("No resolved incidents found")
+type: python
+tags:
+- ml
+comment: Return indicators appears in resolved incidents, and resolved incident ids.
+enabled: true
+args:
+- name: maxIncidents
+  description: Max number of incidents to query.
+  defaultValue: "100000"
+- name: maxIndicators
+  description: Max number of indicators to query.
+  defaultValue: "100000"
+- name: appearsInMinNumberOfIncidents
+  description: Indicator appears in minimum number of incidents.
+  defaultValue: "5"
+- name: query
+  description: The query to fetch the indicators
```

Comment:
The query to fetch the indicators from incident which are "bad"

Reply:
V